File: coding-practice/50 Coding Challenges part 1/C#/problem25.cs
/* Coding challenge #25
Create a function that will receive two arrays of numbers as arguments 
and return an array composed of all the numbers that are either
in the first array or second array but not in both */

using System;
class problem25
{
    static int[] exclusive_array(int[] array1, int[] array2){
        int[] output = new int[array1.Length+array2.Length];
        int count = 0;
        for(int i=0; i<array1.Length; i++){
            bool found = false;
            for(int j=0; j<array2.Length; j++){
                if(array1[i] == array2[j]){
                    found = true;
                    break;
                }
            }
            if(!found){
                output[count] = array1[i];
                count++;
            }

        }

        for(int i=0; i<array2.Length; i++){
            bool found = false;
            for(int j=0; j<array1.Length; j++){
                if(array2[i] == array1[j]){
                    found = true;
                    break;
                }
            }
            if(!found){
                output[count] = array2[i];
                count++;
            }
        }

        Array.Resize(ref output, count);
        return output;
    }

    static void Main(string[] args){
        int[] array1 = {1, 2, 3, 10, 5, 3, 14};
        int[] array2 = {1, 4, 5, 6, 14};
        int[] exclusive = exclusive_array(array1, array2);
        Console.Write(string.Join(", ", exclusive));
    }
}

File: coding-practice/50 Coding Challenges part 1/Python/problem25.py
# Coding challenge #25
# Create a function that will receive 
# two arrays of numbers as arguments 
# and return an array composed of 
# all the numbers that are either 
# in the first array or second array 
# but not in both

array1 = [1, 2, 3, 10, 5, 3, 14]
array2 = [1, 4, 5, 6, 14]
# expected output = [2, 3, 10, 3, 4, 6]

def mutual_elements(arr1, arr2):
    output = []
    for i in arr1:
        found = False
        for j in arr2:
            if i == j:
                found = True
                break
        
        if found == True:
            output.append(i)
    return output

def exclusive_array(arr1, arr2):
    output = []
    mutual = mutual_elements(arr1, arr2)
    for i in mutual:
        arr1.remove(i)
        arr2.remove(i)
    
    for i in arr1:
        output.append(i)
    
    for i in arr2:
        output.append(i)
    
    return output

mutual = mutual_elements(array1, array2)
exclusive = exclusive_array(array1, array2)

print(f'The mutual elements of both arrays are: {mutual}')
print(f'The exclusive array is: {exclusive}')